app/src/behaviors/behavior_to_layer.c

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2020 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#define DT_DRV_COMPAT zmk_behavior_to_layer
+
+#include <device.h>
+#include <drivers/behavior.h>
+#include <logging/log.h>
+
+#include <zmk/keymap.h>
+#include <zmk/behavior.h>
+
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+struct behavior_to_config {};
+struct behavior_to_data {};
+
+static int behavior_to_init(const struct device *dev) { return 0; };
+
+static int to_keymap_binding_pressed(struct zmk_behavior_binding *binding,
+                                     struct zmk_behavior_binding_event event) {
+    LOG_DBG("position %d layer %d", event.position, binding->param1);
+    return zmk_keymap_layer_to(binding->param1);
+}
+
+static int to_keymap_binding_released(struct zmk_behavior_binding *binding,
+                                      struct zmk_behavior_binding_event event) {
+    LOG_DBG("position %d layer %d", event.position, binding->param1);
+    return 0;
+}
+
+static const struct behavior_driver_api behavior_to_driver_api = {
+    .binding_pressed = to_keymap_binding_pressed,
+    .binding_released = to_keymap_binding_released,
+};
+
+static const struct behavior_to_config behavior_to_config = {};
```

Comment:
There's some cruft inherited from other behaviors in our code, but we don't need to carry it forward... If you don't use any config, or data, you can skip defining these types, declaring instances, and just pass `NULL` for these two fields to the device init macro.